core/src/main/java/org/apache/spark/memory/MemoryConsumer.java

```diff
@@ -41,7 +41,7 @@ protected MemoryConsumer(TaskMemoryManager taskMemoryManager, long pageSize, Mem
   }
 
   protected MemoryConsumer(TaskMemoryManager taskMemoryManager) {
```

Comment:
[This constructor](https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/util/collection/Spillable.scala#L29) calls this constructor that specified `ON_HEAP regardless `TaskMemoryManager`'s memory mode.